mlflow/genai/optimize/optimizers/dspy_mipro_optimizer.py

```diff
@@ -80,10 +84,28 @@ def optimize(
 
         adapter = dspy.JSONAdapter()
         with dspy.context(lm=lm, adapter=adapter):
-            dspy_logger = logging.getLogger("dspy")
-            original_level = dspy_logger.level
-            dspy_logger.setLevel(logging.ERROR)
-            try:
+            if not self.optimizer_config.verbose:
+                # Redirect output to devnull if possible, otherwise use StringIO
+                output_sink = io.StringIO()
+                try:
+                    output_sink = open(os.devnull, "w")  # noqa: SIM115
+                except (OSError, IOError):
+                    pass
+
+                with output_sink:
+                    with (
+                        contextlib.redirect_stdout(output_sink),
+                        contextlib.redirect_stderr(output_sink),
+                    ):
```

Comment:
Can we print out these when on error?

Reply:
I don't think the raw dspy logs are useful even when an error occurs. Logs such as instruction candidates or progress bar just overwhelm users who don.